### DISCUSSION TITLE
Multiple Joins / Inner Joins in Dart

### DISCUSSION THREAD

**Discussion by jtkeyva** [supabase/supabase#7341] [Questions]
I'm trying to retrieve a list of posts and the messages within filtered by channel_id. I need the messages to show the username and channel name inside each message. I'm super confused how to do joins, inner joins, multiple joins to output a single data object.

My Data:

Profiles: id, username
Channels: id, name, created_by
Posts: id, title, channel_id, created_by
Messages: id, message, post_id, channel_id, created_by

Output I'm looking for:

Filtered by Channel Food
Posts
 - post: Fruit
 - -message: Banana
 - --created_by: Fred
 - --channel: Food
 - -message: Apple
 - --created_by: Gary
 -  - --channel: Food
 - -message: Orange
 - --created_by: Randy
 -  - --channel: Food

 - post: Vegetables
 - -message: Corn
 - --created_by: Fred
 - --channel: Food
 - -message: Carrot
 - --created_by: Gary
 -  - --channel: Food
 - -message: Potato
 - --created_by: Randy
 -  - --channel: Food

Filtered by Channel: Art
Posts
 - post: Colors
 - -message: Red
 - --created_by: Tina
 - --channel: Art
 - -message: Blue
 - --created_by: Betty
 - --channel: Art
 - -message: Green
 - --created_by: Tina
 - - --channel: Art

 - post: Tools
 - -message: Brush
 - --created_by: Tina
 - --channel: Art
 - -message: Pencil
 - --created_by: Betty
 - --channel: Art
 - -message: Pen
 - --created_by: Tina
 - - --channel: Art

This gives me posts with a list of messages as i need it, but I can't figure out how to get the message to show the Username and the Channel name.

final channelId = '7341';
final res = await supabase
.from('posts')
.select('*, messages(*)
.eq('posts.channel_id', channelId)
.execute()

Please help, this has been torturing me!
Thanks

**Comment by steve-chavez** [comment]
Can you try with:

```dart
final channelId = '7341';
final res = await supabase
.from('posts')
.select('*, messages(message,channels(name),profiles(username))')
.eq('posts.channel_id', channelId)
.execute()
```
